File: leetcode-misc/2559_count_vowel_strings_in_ranges/leetcode2559.cpp
#include <iostream>
#include <vector>
#include <string>
using namespace std;
class Solution {
public:
    bool checkVowel(char c) {
        return (c == 'a' || c == 'e' || c == 'i' || c == 'o' || c == 'u');
    }

    vector<int> vowelStrings(vector<string>& words, vector<vector<int>>& queries) {
        vector<int> result;
        vector<int> prefix = {0};
        int count = 0;

        for (const auto& word : words) {
            if (checkVowel(word.front()) && checkVowel(word.back())) {
                count++;
            }
            prefix.push_back(count);
        }
        for (const auto& query : queries) {
            int l = query[0];
            int r = query[1];
            int ans = prefix[r + 1] - prefix[l];
            result.push_back(ans);
        }
        return result;
    }
};
int main() {
    Solution sol;
    vector<string> words = {"aba", "bcb", "ece", "aa", "e"};
    vector<vector<int>> queries = {{0, 2}, {1, 4}, {1, 1}};
    vector<int> res = sol.vowelStrings(words, queries);
    for (int val : res) {
        cout << val << " ";
    }
}

File: leetcode-misc/2559_count_vowel_strings_in_ranges/leetcode2559.py
from typing import List
class Solution:
    def vowelStrings(self, words: List[str], queries: List[List[int]]) -> List[int]:
        count = 0
        vowel = ['a', 'e', 'i' , 'o', 'u']
        prefix = []
        for word in words:
            count = count + 1 if word[0] in vowel and word[-1] in vowel else count
            prefix.append(count)
        result = list()
        for l,r in queries:
            result.append(prefix[r]-prefix[l-1] if l>=1 else prefix[r])
        return result
if __name__=="__main__":
    sol = Solution()
    print(sol.vowelStrings(["aba","bcb","ece","aa","e"], [[0,2],[1,4],[1,1]]))
    print(sol.vowelStrings(["a","e","i"], [[0,2],[0,1],[2,2]]))
